rust/oak/src/lib.rs

```diff
@@ -239,48 +267,46 @@ impl ChannelPair {
 }
 
 /// Trait encapsulating the operations required for an Oak Node.
-pub trait Node {
+pub trait OakNode {
     fn new() -> Self
     where
         Self: Sized;
     fn invoke(&mut self, method: &str, req: &[u8], out: &mut SendChannelHalf);
 }
 
-thread_local! {
-    static NODE: RefCell<Option<Box<dyn Node>>> = RefCell::new(None);
-}
-
-/// Sets the Oak Node to execute in the current instance.
-///
-/// This function may only be called once, and only from an exported `oak_initialize` function:
-///
-/// ```rust
-/// struct Node;
-///
-/// impl oak::Node for Node {
-///     fn new() -> Self { Node }
-///     fn invoke(&mut self, method: &str, req: &[u8], out: &mut oak::SendChannelHalf) { /* ... */ }
-/// }
-///
-/// #[no_mangle]
-/// pub extern "C" fn oak_initialize() {
-///     oak::set_node::<Node>();
-/// }
-/// ```
-pub fn set_node<T: Node + 'static>() {
+/// Perform an event loop invoking the given node.
+pub fn event_loop<T: OakNode>(mut node: T) {
```

Comment:
May want to have this return `!` (a.k.a. never) to indicate non-termination.
```suggestion
pub fn event_loop<T: OakNode>(mut node: T) -> ! {
```

Reply:
I forgot about that, done.